packages/esm-patient-common-lib/src/dashboards/DashboardExtension.tsx

```diff
@@ -1,24 +1,35 @@
-import React, { useMemo } from 'react';
+import React, { useMemo, useEffect, useState } from 'react';
 import { useLocation } from 'react-router-dom';
 import last from 'lodash-es/last';
 import { ConfigurableLink } from '@openmrs/esm-framework';
 
 export interface DashboardExtensionProps {
-  title: string;
+  title: string | (() => string) | (() => Promise<string>);
   basePath: string;
 }
 
 export const DashboardExtension = ({ title, basePath }: DashboardExtensionProps) => {
   const location = useLocation();
   const navLink = useMemo(() => decodeURIComponent(last(location.pathname.split('/'))), [location.pathname]);
+  const [resolvedTitle, setResolvedTitle] = useState<string>('');
+
+  useEffect(() => {
+    if (typeof title === 'function') {
+      Promise.resolve(title()).then((resolvedValue) => {
+        setResolvedTitle(resolvedValue);
+      });
+    } else {
+      setResolvedTitle(title);
+    }
+  }, [title]);
 
   return (
-    <div key={title}>
+    <div key={resolvedTitle}>
       <ConfigurableLink
-        to={`${basePath}/${encodeURIComponent(title)}`}
-        className={`cds--side-nav__link ${title === navLink && 'active-left-nav-link'}`}
+        to={`${basePath}/${encodeURIComponent(resolvedTitle)}`}
```

Comment:
I'm a little nervous that this won't work properly because we're now assuming that the _translated_ title will resolve to a URL that serves the correct page / component and I don't think that's the case. I think we need to change the `DashboardExtension` to take 3 properties:

1. The base path
2. The translatable title
3. The URI fragment used for this component.

Basically, the thing used in `to` and `className` should be the URI fragment (so, non-translated) and the translated title should _only_ be used as the content of the link tag.

Reply:
Basically, the URL generated for a page in OpenMRS should not depend on the user's language.